### ISSUE TITLE
Problem Spyder check updates

### ISSUE THREAD

**Issue by gs112145** [spyder-ide/spyder#20779]
### What steps will reproduce the problem?

I activated my environment with 
my_environment/Scripts/activate.
Then I started Sypder and I've got the following error.
This happened every time within this environment.

### Traceback
```python-traceback
Traceback (most recent call last):
  File "z:\tensorflow\mytensorflow_env\lib\site-packages\spyder\plugins\application\container.py", line 331, in _check_updates_ready
    msg = header + content + footer
UnboundLocalError: local variable 'content' referenced before assignment
```

## Versions

* Spyder version: 5.4.2  (pip)
* Python version: 3.9.6 64-bit
* Qt version: 5.15.2
* PyQt5 version: 5.15.9
* Operating System: Windows 10

### Dependencies

```
# Mandatory:
atomicwrites >=1.2.0          :  1.4.0 (OK)
chardet >=2.0.0               :  5.1.0 (OK)
cloudpickle >=0.5.0           :  2.2.1 (OK)
cookiecutter >=1.6.0          :  2.1.1 (OK)
diff_match_patch >=20181111   :  20200713 (OK)
intervaltree >=3.0.2          :  3.1.0 (OK)
IPython >=7.31.1;<9.0.0       :  8.11.0 (OK)
jedi >=0.17.2;<0.19.0         :  0.18.2 (OK)
jellyfish >=0.7               :  0.10.0 (OK)
jsonschema >=3.2.0            :  4.17.3 (OK)
keyring >=17.0.0              :  23.13.1 (OK)
nbconvert >=4.0               :  7.2.10 (OK)
numpydoc >=0.6.0              :  1.5.0 (OK)
paramiko >=2.4.0              :  3.1.0 (OK)
parso >=0.7.0;<0.9.0          :  0.8.3 (OK)
pexpect >=4.4.0               :  4.8.0 (OK)
pickleshare >=0.4             :  0.7.5 (OK)
psutil >=5.3                  :  5.9.4 (OK)
pygments >=2.0                :  2.14.0 (OK)
pylint >=2.5.0;<3.0           :  2.17.1 (OK)
pylint_venv >=2.1.1           :  3.0.1 (OK)
pyls_spyder >=0.4.0           :  0.4.0 (OK)
pylsp >=1.7.1;<1.8.0          :  1.7.1 (OK)
pylsp_black >=1.2.0           :  1.2.1 (OK)
qdarkstyle >=3.0.2;<3.1.0     :  3.0.3 (OK)
qstylizer >=0.2.2             :  0.2.2 (OK)
qtawesome >=1.2.1             :  1.2.3 (OK)
qtconsole >=5.4.0;<5.5.0      :  5.4.1 (OK)
qtpy >=2.1.0                  :  2.3.0 (OK)
rtree >=0.9.7                 :  1.0.1 (OK)
setuptools >=49.6.0           :  67.5.1 (OK)
sphinx >=0.6.6                :  6.1.3 (OK)
spyder_kernels >=2.4.2;<2.5.0 :  2.4.2 (OK)
textdistance >=4.2.0          :  4.5.0 (OK)
three_merge >=0.1.1           :  0.1.1 (OK)
watchdog >=0.10.3             :  3.0.0 (OK)
zmq >=22.1.0                  :  25.0.0 (OK)

# Optional:
cython >=0.21                 :  0.29.33 (OK)
matplotlib >=3.0.0            :  3.7.1 (OK)
numpy >=1.7                   :  1.24.2 (OK)
pandas >=1.1.1                :  1.5.3 (OK)
scipy >=0.17.0                :  1.10.1 (OK)
sympy >=0.7.3                 :  1.11.1 (OK)
```

### Environment

<details><summary>Environment</summary>

```
Package            Version
------------------ ------------
accelerate         0.18.0
aiohttp            3.8.4
aiosignal          1.3.1
anvil-app-server   1.7.6
anvil-uplink       0.3.42
asttokens          2.2.1
async-timeout      4.0.2
attrs              22.2.0
backcall           0.2.0
certifi            2022.12.7
charset-normalizer 2.1.1
click              8.1.3
cloudpickle        2.2.1
colorama           0.4.6
comm               0.1.3
datasets           2.11.0
debugpy            1.6.6
decorator          5.1.1
dill               0.3.6
entrypoints        0.4
evaluate           0.4.0
executing          1.2.0
filelock           3.9.0
frozenlist         1.3.3
fsspec             2023.3.0
future             0.18.3
happytransformer   2.4.1
huggingface-hub    0.13.3
idna               3.4
ipykernel          6.22.0
ipython            8.12.0
ipywidgets         8.0.6
jedi               0.18.2
Jinja2             3.1.2
jupyter_client     7.4.9
jupyter_core       5.3.0
jupyterlab-widgets 3.0.7
MarkupSafe         2.1.2
matplotlib-inline  0.1.6
mpmath             1.2.1
multidict          6.0.4
multiprocess       0.70.14
nest-asyncio       1.5.6
networkx           3.0
numpy              1.24.1
packaging          23.0
pandas             1.5.3
parso              0.8.3
pickleshare        0.7.5
Pillow             9.3.0
pip                23.0.1
platformdirs       3.2.0
progressbar2       4.2.0
prompt-toolkit     3.0.38
protobuf           4.22.1
psutil             5.9.4
pure-eval          0.2.2
pyarrow            11.0.0
pychrome           0.2.3
Pygments           2.14.0
python-dateutil    2.8.2
python-utils       3.5.2
pytz               2023.3
pywin32            306
PyYAML             6.0
pyzmq              25.0.2
regex              2023.3.23
requests           2.28.1
responses          0.18.0
sentencepiece      0.1.97
setuptools         65.5.0
six                1.16.0
spyder-kernels     2.4.2
stack-data         0.6.2
sympy              1.11.1
tokenizers         0.13.2
torch              2.0.0+cu118
torchaudio         2.0.1+cu118
torchvision        0.15.1+cu118
tornado            6.2
tqdm               4.65.0
traitlets          5.9.0
transformers       4.27.4
typing_extensions  4.4.0
urllib3            1.26.13
wcwidth            0.2.6
websocket-client   1.5.1
wheel              0.40.0
widgetsnbextension 4.0.7
ws4py              0.5.1
xxhash             3.2.0
yarl               1.8.2

```
</details>

**Comment by dalthviz** [comment]
Hi @gs112145  thank you for the feedback! This seems like a duplicate of #20406 To prevent this you will need to either deactivate the check for updates at startup option or update Spyder to the latest release (5.4.3). The actual fix was released with Spyder 5.4.3 so hopefully if you update to 5.4.3 then when we release a new version you will get the actual message indicating an update is available instead of the error message related with the construction of the dialog.

Let us know if the info above helps!

**Comment by dalthviz** [comment]
Closing as a duplicate of https://github.com/spyder-ide/spyder/issues/20406
